notebooks/textbook/Recursive_Quantum_Approximate_Optimization_Algorithm.ipynb

```diff
@@ -0,0 +1,579 @@
+    "\n",
+    "\n",
+    "* The ``custom`` strategy allows the user to define the number of eliminations to be performed at each step. This defined by the parameter ``steps``. If the parameter is set as an integer, the algorithm will use this value as the number of qubits to be eliminated at each step. Alternatively, it is possible to pass a list, which specifies the number of qubits to be eliminated at each step. For ``steps = 1``, the algorithm reduces to the original form of RQAOA presented in [1].\n",
+    "\n",
+    "* The ``adaptive`` strategy adaptively selects how many qubits to eliminate at each step. The maximum number of allowed eliminations is given by the parameter ``n_max``. At each step, the algorithm selects the top ``n_max+1`` expectation values (ranked in magnitude), computes the mean among them, and uses the ones lying above it for qubit elimination. This corresponds to a maximum of ``n_max`` possible elimination per step. For ``n_max= 1``, the algorithm reduces to the original form of RQAOA presented in [1].\n",
+    "\n",
+    "**NOTE**: The specific performance of these generalizations is currently under investigation. In particular, the development of Adaptive RQAOA is associated with an internal research project at Entropica Labs to be released publicly in the near future [4]. We make these strategies already available to the community in order to strengthen the exploration of more complex elimination schemes for RQAOA, beyond its original formulation [1]."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "61270353-74c9-4163-a663-5c89d18976fc",
+   "metadata": {},
+   "source": [
+    "# References"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6534a736-093b-44d3-95b6-7a8fa309c2f8",
+   "metadata": {},
+   "source": [
+    "[1] S. Bravyi, A. Kliesch, R. Koenig, and E. Tang, [Physical Review Letters 125, 260505 (2020)](https://journals.aps.org/prl/abstract/10.1103/PhysRevLett.125.260505) \\\n",
+    "[2] S. Bravyi, A. Kliesch, R. Koenig, and E. Tang, [(2020), 10.22331/q-2022-03-30-678](https://quantum-journal.org/papers/q-2022-03-30-678/) \\\n",
+    "[3] D. J. Egger, J. Marecek, and S. Woerner, [Quantum 5, 479 (2021)](https://doi.org/10.22331/q-2021-06-17-479) \\\n",
+    "[4] E. I. Rodríguez Chiacchio, V. Sharma, E. Munro (Work in progress) "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "3870b298-fc8e-44ab-9a49-bbd861b1a055",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%load_ext autoreload\n",
+    "%autoreload 2"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "id": "f1b38648-393a-4974-af43-a2c7d960fb17",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "try:\n",
+    "    import openqaoa\n",
+    "except ImportError:\n",
+    "    !pip -q install openqaoa"
```

Comment:
Looks like this requires python 3.8 or higher. We should hold off merging until Braket has deprecated use of Python 3.7 but PRs exist and we plan to.